### PR TITLE
Add more opcodes in SMSG

### PR DESCRIPTION
**Pull request by diamondo25**
In addition to this, rename SMSG_YELLOW_MESSAGE to SMSG_SET_WEEK_EVENT_MESSAGE (GMS-like)

### CODE REVIEW

File: src/channel_server/smsg_header.hpp [hunk 1]
```diff
@@ -26,38 +26,52 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
-#define SMSG_REPORT_MORE 0x2a // More report functionality
+// 0x29 does not exist?
+#define SMSG_CLAIM_RESULT 0x2a
+#define SMSG_CLAIM_SERVER_AVAILABLE_TIME 0x2b // byte startHour, byte endHour
+#define SMSG_CLAIM_SERVER_STATUS_CHANGED 0x2c // bool isAvailable
+#define SMSG_SET_TAMINGMOB_INFO 0x2d // int characterid, int level, int exp, int fatigue, bool isLevelUp (broadcast to map)
 #define SMSG_QUEST_COMPLETED 0x2e
 #define SMSG_PLAYER_STORE_MESSAGE 0x2f
 #define SMSG_SKILLBOOK 0x30
+// 0x31 and 0x32 are not doing anything other than reading 2 bytes
```
Comment (LazyBui) [inline src/channel_server/smsg_header.hpp]: 0x33?

File: src/channel_server/smsg_header.hpp [hunk 2]
```diff
@@ -26,38 +26,52 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
 #define SMSG_REPORT_USER 0x34
 #define SMSG_REACHED_TRADE_MESOS_LIMIT 0x36 // "Players that are Level 15 and below may only trade 1 million mesos per day. You have reached the limit today, please try again tomorrow"
+// 0x38 is empty
```
Comment (LazyBui) [inline src/channel_server/smsg_header.hpp]: 0x37, 0x39?

File: src/channel_server/smsg_header.hpp [hunk 3]
```diff
@@ -26,38 +26,52 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
 #define SMSG_PORTAL_ACTION 0x40
 #define SMSG_MESSAGE 0x41
+#define SMSG_INCUBATOR 0x42
 #define SMSG_OWL_OF_MINERVA 0x43
+// 0x44 does nothing
 #define SMSG_WEDDING_PROPOSAL_WHISHLIST 0x45 // Holds both whishlist and proposal
 #define SMSG_WEDDING 0x46 // Holds wedding and proposal things like "You are now engaged."...
 #define SMSG_WEDDING_MORE 0x47 // Holds wedding stuff too.....
-#define SMSG_YELLOW_MESSAGE 0x4a // Used by bosses like Stumpy and Mano
+#define SMSG_USE_CASH_PET_FOOD 0x49 // if (failed) { bool true } else { bool false, byte pet index }
```
Comment (LazyBui) [inline src/channel_server/smsg_header.hpp:68]: 0x48?

File: src/channel_server/smsg_header.hpp [hunk 4]
```diff
@@ -122,12 +136,18 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+#define SMSG_NOTIFY_HP_DAMAGE_BY_FIELD 0xc2 // When the field gives you, for example, cold damage, send this opcode + amount of damage
+// 0xc3 is not implemented
 #define SMSG_BUBBLE 0xc4
-#define SMSG_SOUND 0xc6
+#define SMSG_EVENT_SOUND 0xc5 // Play event (field) sound. Sounds under Sound.wz\Field.img\ are accepted
+#define SMSG_SOUND 0xc6 // Play minigame sound
+#define SMSG_MAKER 0xc7 // Creating items and such
 #define SMSG_EVENT_BOARD 0xc9 // Korean event board :D
+// 0xca && 0xcb seem to be UI open/close related
+// 0xcc is not implemented
```
Comment (LazyBui) [inline src/channel_server/smsg_header.hpp]: 0xcd?

**Review by LazyBui — APPROVED** [review]
Happy to merge. I'm a bit confused on the gaps though. :p

**Comment by diamondo25** [comment]
@LazyBui Its a bit more than expected; we were missing a lot of opcodes mainly because of:
- Map-based opcodes
- Family opcodes
- Stuff that wasnt needed anyway (MTS, cashshop things, adminshop, etc)

**Comment by diamondo25** [comment]
@LazyBui I think this is ready for review, as its (i think) 95% of all packet headers (almost enough to start using an enum). The other 5% doesnt seem to be handled anywhere (maybe debug opcodes or something)

**Pull request merged**